Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,39 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        return lhs.alert == rhs.alert
+            && lhs.badge == rhs.badge
+            && lhs.sound == rhs.sound
+            && lhs.contentAvailable == rhs.contentAvailable
+            && lhs.categoryIdentifier == rhs.categoryIdentifier
+            && NSDictionary(dictionary: lhs.userInfo) == NSDictionary(dictionary: rhs.userInfo)
+    }
+    
+    public init?(remoteNotification: [AnyHashable : Any]?) {
+        guard let remoteNotification = remoteNotification as! [String : Any]? else {
             return nil
         }
         
         // Parse the notification payload.
         // See https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html for more information on expected values.
         alert = Alert(remoteNotification: remoteNotification)
         
-        let apnsDictionary = remoteNotification[APSServiceKey] as? [String : AnyObject]
-        badge = apnsDictionary?[badgeKey] as? Int
-        sound = apnsDictionary?[soundKey] as? String
+        let apnsDictionary = remoteNotification[apsServiceKey] as! [String : Any]?
```

Comment:
I'd say return an error for at least two reasons:
1) These crashes aren't catchable by the caller, so you're making the choice for them as to what behavior their app wants in the face of failure.
2) This is not a response that you control (Apple does), so if it does start failing, you can't fix the underlying issue. If you have a catchable error though, you can at least mitigate it.

Generally, think of `!` as an assert, and asserts should be used to validate *your own* logic. External input should be validated via standard error handling.